tests/phpunit/test-wc-stripe-sub-initial.php

```diff
@@ -0,0 +1,217 @@
+ *
+ * There are a few methods that need to be mocked in the class WC_Stripe_Subs_Compat, which is
+ * why that class is mocked even though the method under test is part of that class.
+ *
+ * @package WooCommerce_Stripe/Classes/WC_Stripe_Subscription_Initial_Test
+ */
+
+/**
+ * WC_Stripe_Subscription_Initial_Test
+ */
+class WC_Stripe_Subscription_Initial_Test extends WP_UnitTestCase {
+	/**
+	 * System under test, and a mock object with some methods mocked for testing
+	 *
+	 * @var PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $wc_stripe_subs_compat;
+
+	/**
+	 * The statement descriptor we'll use in a test.
+	 *
+	 * @var string
+	 */
+	private $statement_descriptor;
+
+	/**
+	 * Sets up things all tests need.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->wc_stripe_subs_compat = $this->getMockBuilder( 'WC_Stripe_Subs_Compat' )
+			->disableOriginalConstructor()
+			->setMethods( array( 'prepare_source', 'has_subscription' ) )
+			->getMock();
+
+		// Mocked in order to get metadata[payment_type] = recurring in the HTTP request.
+		$this->statement_descriptor = 'This is a statement descriptor.';
+		update_option(
+			'woocommerce_stripe_settings',
+			array(
+				'statement_descriptor' => $this->statement_descriptor,
+			)
+		);
+	}
+
+	/**
+	 * Tears down the stuff we set up.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+		delete_option( 'woocommerce_stripe_settings' );
+	}
+
+	/**
+	 * Tests whether the initial payment succeeds and includes the `setup_future_usage` parameter.
+	 *
+	 * 1. Several things are set up or mocked.
+	 * 2. A function that will mock an HTTP response for the payment_intents is created.
+	 * 3. That same function has some assertions about the things we send to the
+	 * payments_intents endpoint.
+	 * 4. The function under test - `process_payment` - is called.
+	 * 5. More assertions are made.
+	 */
+	public function test_initial_intent_parameters() {
+		$initial_order        = WC_Helper_Order::create_order();
+		$order_id             = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $initial_order->id : $initial_order->get_id();
+		$amount               = 50;
+		$stripe_amount        = WC_Stripe_Helper::get_stripe_amount( $amount );
+		$currency             = strtolower( WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $initial_order->get_order_currency() : $initial_order->get_currency() );
+		$customer             = 'cus_123abc';
+		$source               = 'src_123abc';
+		$statement_descriptor = WC_Stripe_Helper::clean_statement_descriptor( $this->statement_descriptor );
+		$intents_api_endpoint = 'https://api.stripe.com/v1/payment_intents';
+		$urls_used            = array();
+
+		if ( WC_Stripe_Helper::is_wc_lt( '3.0' ) ) {
+			$initial_order->payment_method = 'stripe';
+		} else {
+			$initial_order->set_payment_method( 'stripe' );
+			$initial_order->save();
+		}
+
+		// Arrange: Mock prepare_order_source() so that we have a customer and source.
```

Comment:
I think you mean `prepare_source `, not `prepare_order_source` in the comment.

Reply:
Yes, fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/962/commits/781d8ac344b775fc7ce56428f54edf3fc8b5da26